allensdk/brain_observatory/behavior/running_processing.py

```diff
@@ -7,13 +7,9 @@
 
 
 def calc_deriv(x, time):
-    dx = np.diff(x)
-    dt = np.diff(time)
-    dxdt_rt = np.hstack((np.nan, dx / dt))
-    dxdt_lt = np.hstack((dx / dt, np.nan))
-    dxdt = np.vstack((dxdt_rt, dxdt_lt))
-    dxdt = np.nanmean(dxdt, axis=0)
-    return dxdt
+    dx = np.diff(x, prepend=np.nan)
+    dt = np.diff(time, prepend=np.nan)
+    return dx / dt
```

Comment:
this function returns something different now. Is that desired?
```
In [12]: def calc_deriv(x, time): 
    ...:     dx = np.diff(x) 
    ...:     dt = np.diff(time) 
    ...:     dxdt_rt = np.hstack((np.nan, dx / dt)) 
    ...:     dxdt_lt = np.hstack((dx / dt, np.nan)) 
    ...:     dxdt = np.vstack((dxdt_rt, dxdt_lt)) 
    ...:     dxdt = np.nanmean(dxdt, axis=0) 
    ...:     return dxdt 
    ...:                                                                          

In [13]: x = np.arange(10)                                                        

In [14]: x = np.power(np.arange(10), 2)                                           

In [15]: x                                                                        
Out[15]: array([ 0,  1,  4,  9, 16, 25, 36, 49, 64, 81])

In [16]: t = np.arange(x.size)                                                    

In [17]:                                                                          

In [17]: calc_deriv(x, t)                                                         
Out[17]: array([ 1.,  2.,  4.,  6.,  8., 10., 12., 14., 16., 17.])

In [18]: def calc_deriv2(x, time): 
    ...:     dx = np.diff(x, prepend=np.nan) 
    ...:     dt = np.diff(time, prepend=np.nan) 
    ...:     return dx / dt 
    ...:                                                                          

In [19]: calc_deriv2(x, t)                                                        
Out[19]: array([nan,  1.,  3.,  5.,  7.,  9., 11., 13., 15., 17.])
```

Reply:
It was intentional to change the output. Using the old derivative calculation caused differences between SDK and VBA output. I think the purpose of the old algorithm was just to avoid nans. Although the approach seems unconventional to me (mean of 2 derivatives where 1 is prepended with nan and the other appended with nan?)

Reply:
:heavy_check_mark: